platforms/raspi/pwm_pin.go

```diff
@@ -9,19 +9,19 @@ import (
 	"gobot.io/x/gobot/sysfs"
 )
 
-const piBlasterPeriod = 10000000
-
 // PWMPin is the Raspberry Pi implementation of the PWMPinner interface.
 // It uses Pi Blaster.
 type PWMPin struct {
-	pin string
-	dc  uint32
+	pin    string
+	dc     uint32
+	period *uint32
 }
 
 // NewPwmPin returns a new PWMPin
-func NewPWMPin(pin string) *PWMPin {
+func NewPWMPin(pin string, period *uint32) *PWMPin {
```

Comment:
The `NewPWMPin()` function signature is the same across all the Adaptors that support it, so you probably should not add another param here.

Reply:
Thanks. I'll try other way around it.